src/app/systelab-components/datepicker/datepicker.component.ts

```diff
@@ -237,7 +242,30 @@ export class Datepicker implements OnInit, AfterViewInit, DoCheck, OnDestroy {
 		}
 	}
 
-	public onKeyDown(event: KeyboardEvent) {
+	public formatDate(date: string ): string {
+		let dateTmp: string = date.trim();
+
+		if (dateTmp != null) {
+			if ((dateTmp.length === 8 || dateTmp.length === 10)
+				&& (dateTmp.lastIndexOf('/')  > 0 || dateTmp.lastIndexOf('-') > 0 || dateTmp.lastIndexOf('.') > 0)) {
+				dateTmp = dateTmp.replace('/', '')
+					.replace('-', '')
+					.replace('.', '');
+			}
+
+			if (dateTmp.length === 6) {
+				dateTmp = dateTmp.substring(0, 2) + '/' + dateTmp.substring(2, 4) + '/20'
```

Comment:
20 hardcoded?

Reply:
In our case, most of the time a year with 2 digits refers to year 20XX (create a visit, result, tp, billing, etc...) only the date of birth refers to year 19XX. So statistically it is more significative the 20